xds/src/test/java/io/grpc/xds/LookasideLbTest.java

```diff
@@ -325,485 +370,613 @@ XdsClient createXdsClient() {
+    }
+    for (LocalityLbEndpoints localityLbEndpoints : clusterLoadAssignment.getEndpointsList()) {
+      endpointUpdateBuilder.addLocalityLbEndpoints(
+          EnvoyProtoData.Locality.fromEnvoyProtoLocality(
+              localityLbEndpoints.getLocality()),
+          EnvoyProtoData.LocalityLbEndpoints.fromEnvoyProtoLocalityLbEndpoints(
+              localityLbEndpoints));
+    }
+    return endpointUpdateBuilder.build();
+  }
 
-    LocalityStore localityStore = Iterables.getOnlyElement(localityStores);
-    LoadReportClient loadReportClient = Iterables.getOnlyElement(loadReportClients);
-    verify(localityStore, never()).reset();
-    verify(loadReportClient, never()).stopLoadReporting();
-    assertThat(channel.isShutdown()).isFalse();
+  private void handleResolvedAddresses(XdsConfig xdsConfig) {
+    ResolvedAddresses.Builder resolvedAddressBuilder = ResolvedAddresses.newBuilder()
+        .setAddresses(ImmutableList.<EquivalentAddressGroup>of())
+        .setLoadBalancingPolicyConfig(xdsConfig);
+    if (withXdsClientPoolAttributes) {
+      resolvedAddressBuilder.setAttributes(
+          Attributes.newBuilder().set(XdsAttributes.XDS_CLIENT_POOL,
+              xdsClientPoolFromResolveAddresses).build());
+    }
+    lookasideLb.handleResolvedAddresses(resolvedAddressBuilder.build());
+  }
 
-    lookasideLb.shutdown();
+  private int versionIno;
+  private int nonce;
+
+  private void receiveEndpointUpdate(ClusterLoadAssignment clusterLoadAssignment) {
+    if (withBootstrap) {
+      responseObservers.peekLast().onNext(
+          buildDiscoveryResponse(
+              String.valueOf(versionIno++),
+              ImmutableList.of(Any.pack(clusterLoadAssignment)),
+              XdsClientImpl.ADS_TYPE_URL_EDS,
+              String.valueOf(nonce++)));
+    } else if (withXdsClientPoolAttributes) {
+      endpointWatchers.get(clusterLoadAssignment.getClusterName())
+          .onEndpointChanged(getEndpointUpdateFromClusterAssignment(clusterLoadAssignment));
+    }
+  }
 
-    verify(localityStore).reset();
-    verify(loadReportClient).stopLoadReporting();
-    assertThat(channel.isShutdown()).isTrue();
+  private void assertLatestConnectivityState(ConnectivityState expectedState) {
+    verify(helper, atLeastOnce()).updateBalancingState(
+        connectivityStateCaptor.capture(), pickerCaptor.capture());
+    assertThat(connectivityStateCaptor.getValue()).isEqualTo(expectedState);
+  }
+
+  private void assertLatestSubchannelPicker(Subchannel expectedSubchannelToPick) {
+    assertLatestConnectivityState(READY);
+    assertThat(
+            pickerCaptor.getValue().pickSubchannel(mock(PickSubchannelArgs.class)).getSubchannel())
+        .isEqualTo(expectedSubchannelToPick);
+  }
+
+  /**
+   * Matcher of ResolvedAddresses for round robin load balancer based on the set of backends.
+   */
+  private static class RoundRobinBackendsMatcher implements ArgumentMatcher<ResolvedAddresses> {
```

Comment:
Does the order of `EquivalentAddressGroup`s in `ResolvedAddresses` matter? Anyway, `EndpointUpdate` preserves the order of addresses sent from server side. That is, you are able to assert directly on the address list with `containsExactly`. But really, this depends on if the order matters to the LB policy semantics.

Reply:
The order of address list for round robin policy matters.

Reply:
Then why still use this `RoundRobinBackendsMatcher` that does not care ordering of backends in `ResolvedAddresses`?

Reply:
I see what you mean. I didn't test that aspect because I was not concerned about the order. Now adding it.